user-interface/src/main/java/life/qbic/datamanager/views/projects/project/ontology/OntologyLookupComponent.java

```diff
@@ -0,0 +1,235 @@
+
+import com.vaadin.flow.component.ComponentEvent;
+import com.vaadin.flow.component.ComponentEventListener;
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.component.dependency.JsModule;
+import com.vaadin.flow.component.grid.Grid;
+import com.vaadin.flow.component.grid.GridVariant;
+import com.vaadin.flow.component.grid.dataview.GridLazyDataView;
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.component.icon.Icon;
+import com.vaadin.flow.component.icon.VaadinIcon;
+import com.vaadin.flow.component.textfield.TextField;
+import com.vaadin.flow.data.provider.SortDirection;
+import com.vaadin.flow.data.value.ValueChangeMode;
+import com.vaadin.flow.spring.annotation.SpringComponent;
+import com.vaadin.flow.spring.annotation.UIScope;
+import com.vaadin.flow.theme.lumo.LumoUtility.IconSize;
+import java.io.Serial;
+import java.util.Arrays;
+import java.util.List;
+import life.qbic.datamanager.views.general.Card;
+import life.qbic.datamanager.views.general.PageArea;
+import life.qbic.datamanager.views.general.Tag;
+import life.qbic.projectmanagement.application.OntologyTermInformationService;
+import life.qbic.projectmanagement.application.SortOrder;
+import life.qbic.projectmanagement.domain.model.Ontology;
+import life.qbic.projectmanagement.domain.model.experiment.vocabulary.OntologyClassDTO;
+import org.springframework.beans.factory.annotation.Autowired;
+
+
+/**
+ * Ontology Lookup Component
+ * <p>
+ * The ontology lookup component is a {@link PageArea} component, which enables a user to look up
+ * the ontologies supported within the data manager. It allows the user to provide a string based
+ * search term and get results with the correct IRI for the ontology terms.
+ */
+@SpringComponent
+@UIScope
+@JsModule("./javascript/copytoclipboard.js")
+public class OntologyLookupComponent extends PageArea {
+
+  @Serial
+  private static final long serialVersionUID = -1777819501917841723L;
+  private final TextField searchField = new TextField();
+  private final Div ontologyGridSection = new Div();
+  private GridLazyDataView<OntologyClassDTO> ontologyGridLazyDataView;
+  private String searchTerm = "";
+  private final Span foundResults = new Span();
+  private final transient OntologyTermInformationService ontologyTermInformationService;
+  private static final int ONTOLOGY_SEARCH_LOWER_LIMIT = 2;
+
+  public OntologyLookupComponent(
+      @Autowired OntologyTermInformationService ontologyTermInformationService) {
+    requireNonNull(ontologyTermInformationService);
+    this.ontologyTermInformationService = ontologyTermInformationService;
+    Span title = new Span("Ontology Search/Lookup");
+    title.addClassName("title");
+    add(title);
+    Span description = new Span(
+        "You can search here for ontology terms from over 20 different ontologies.");
+    add(description);
+    addSearchEventListener(this::updateShownResults);
```

Comment:
why not add the listener to the textfield directly?

Reply:
I thought it's better to have a dedicated method to keep the constructor slim. If you think it makes more sense to initialize this in the constructor then i'll change it 🤔